lib/activerecord_slotted_counters/adapters/mysql_upsert.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module ActiveRecordSlottedCounters
+  module Adapters
+    class MysqlUpsert
+      attr_reader :klass, :current_adapter_name
+
+      def initialize(klass, current_adapter_name)
+        @klass = klass
+        @current_adapter_name = current_adapter_name
```

Comment:
We don't this to be a part of the adapter state; we only use it in the `apply?` method, so let's pass current adapter as an argument to `#apply?` instead.

Reply:
thanks
got it!
done!